<a href="https://colab.research.google.com/github/kritp144/Sales-Prediction/blob/main/SalesDataPart2_0.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [73]:
# Import panda 
import pandas as pd

# Data file assigned
food_sales_data = '/content/drive/MyDrive/DATA SCIENCE/CODES/project 1/sales_predictions.csv'

# Upload the datafile to panda
foodsales_df = pd.read_csv(food_sales_data)

#Examine the first 5 rows
foodsales_df.head()

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
0           FDA15         9.30          Low Fat         0.016047   
1           DRC01         5.92          Regular         0.019278   
2           FDN15        17.50          Low Fat         0.016760   
3           FDX07        19.20          Regular         0.000000   
4           NCD19         8.93          Low Fat         0.000000   

               Item_Type  Item_MRP Outlet_Identifier  \
0                  Dairy  249.8092            OUT049   
1            Soft Drinks   48.2692            OUT018   
2                   Meat  141.6180            OUT049   
3  Fruits and Vegetables  182.0950            OUT010   
4              Household   53.8614            OUT013   

   Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
0                       1999      Medium               Tier 1   
1                       2009      Medium               Tier 3   
2                       1999      Medium               Tier 1   
3                       1998         NaN               Tier 3   
4                       1987        High               Tier 3   

         Outlet_Type  Item_Outlet_Sales  
0  Supermarket Type1          3735.1380  
1  Supermarket Type2           443.4228  
2  Supermarket Type1          2097.2700  
3      Grocery Store           732.3800  
4  Supermarket Type1           994.7052

The Data presented above is in its raw form therefore, it requires to be cleaned before performing any analytics. Here below, we will go through the precess of cleaning the data.

In [74]:
# Determine the total number of rows and columns
foodsales_df.shape


(8523, 12)

In [75]:
# Determine the Data Type of each variable
foodsales_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Weight                7060 non-null   float64
 2   Item_Fat_Content           8523 non-null   object 
 3   Item_Visibility            8523 non-null   float64
 4   Item_Type                  8523 non-null   object 
 5   Item_MRP                   8523 non-null   float64
 6   Outlet_Identifier          8523 non-null   object 
 7   Outlet_Establishment_Year  8523 non-null   int64  
 8   Outlet_Size                6113 non-null   object 
 9   Outlet_Location_Type       8523 non-null   object 
 10  Outlet_Type                8523 non-null   object 
 11  Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 799.2+ KB


In [76]:
# Check for duplicates
sum(foodsales_df.duplicated())

0

In [77]:
# Identify all missing values in the columns
foodsales_df.isna().sum()


Item_Identifier                 0
Item_Weight                  1463
Item_Fat_Content                0
Item_Visibility                 0
Item_Type                       0
Item_MRP                        0
Outlet_Identifier               0
Outlet_Establishment_Year       0
Outlet_Size                  2410
Outlet_Location_Type            0
Outlet_Type                     0
Item_Outlet_Sales               0
dtype: int64

We have Identified that Item Weight has 1463 values missing nad Outlet size has 2410 values missing


# STEP 1: Dealing with Item Weight
 

Item weight has 1463 missing spaces which accumulated for about 17% of our data. Deleting that much amount of data could skew our results one way or another.

However, the Item Weight has no correlation to Sales and therefore eliminating that column would maintain the intergrity of our data.

In [78]:
# Delete the Item Weight Coulmn
foodsales_df.drop(columns= 'Item_Weight', inplace= True)

In [79]:
# Check for confirmation
foodsales_df.isna().sum()


Item_Identifier                 0
Item_Fat_Content                0
Item_Visibility                 0
Item_Type                       0
Item_MRP                        0
Outlet_Identifier               0
Outlet_Establishment_Year       0
Outlet_Size                  2410
Outlet_Location_Type            0
Outlet_Type                     0
Item_Outlet_Sales               0
dtype: int64

# STEP 2: Dealing with Outlet Size data


Outlet Size has 2410 missing spaces which accumulated for about 28% of our data. Deleting that much amount of data could skew our results one way or another.

However, the Oulet Size  has no correlation to Sales and therefore eliminating that column would maintain the intergrity of our data.

In [80]:
# Delete the Outlet Size Coulmn
foodsales_df.drop(columns= 'Outlet_Size', inplace= True)


In [81]:
# Check for confirmation
foodsales_df.isna().sum()



Item_Identifier              0
Item_Fat_Content             0
Item_Visibility              0
Item_Type                    0
Item_MRP                     0
Outlet_Identifier            0
Outlet_Establishment_Year    0
Outlet_Location_Type         0
Outlet_Type                  0
Item_Outlet_Sales            0
dtype: int64

In [82]:
# Round up Item MRP to 2 decimal places
foodsales_df['Item_MRP'] =  foodsales_df['Item_MRP'].round(2)

# Round up Item MRP to 2 decimal places
foodsales_df['Item_Visibility'] =  foodsales_df['Item_Visibility'].round(3)

In [83]:
# Confirm that it got rounded
foodsales_df

Item_Identifier Item_Fat_Content  Item_Visibility              Item_Type  \
0              FDA15          Low Fat            0.016                  Dairy   
1              DRC01          Regular            0.019            Soft Drinks   
2              FDN15          Low Fat            0.017                   Meat   
3              FDX07          Regular            0.000  Fruits and Vegetables   
4              NCD19          Low Fat            0.000              Household   
...              ...              ...              ...                    ...   
8518           FDF22          Low Fat            0.057            Snack Foods   
8519           FDS36          Regular            0.047           Baking Goods   
8520           NCJ29          Low Fat            0.035     Health and Hygiene   
8521           FDN46          Regular            0.145            Snack Foods   
8522           DRG01          Low Fat            0.045            Soft Drinks   

      Item_MRP Outlet_Identifier  Outlet_Establishment_Year  \
0       249.81            OUT049                       1999   
1        48.27            OUT018                       2009   
2       141.62            OUT049                       1999   
3       182.10            OUT010                       1998   
4        53.86            OUT013                       1987   
...        ...               ...                        ...   
8518    214.52            OUT013                       1987   
8519    108.16            OUT045                       2002   
8520     85.12            OUT035                       2004   
8521    103.13            OUT018                       2009   
8522     75.47            OUT046                       1997   

     Outlet_Location_Type        Outlet_Type  Item_Outlet_Sales  
0                  Tier 1  Supermarket Type1          3735.1380  
1                  Tier 3  Supermarket Type2           443.4228  
2                  Tier 1  Supermarket Type1          2097.2700  
3                  Tier 3      Grocery Store           732.3800  
4                  Tier 3  Supermarket Type1           994.7052  
...                   ...                ...                ...  
8518               Tier 3  Supermarket Type1          2778.3834  
8519               Tier 2  Supermarket Type1           549.2850  
8520               Tier 2  Supermarket Type1          1193.1136  
8521               Tier 3  Supermarket Type2          1845.5976  
8522               Tier 1  Supermarket Type1           765.6700  

[8523 rows x 10 columns]

In [84]:
# Check for similar values in Item Type
foodsales_df['Item_Type'].value_counts()

Fruits and Vegetables    1232
Snack Foods              1200
Household                 910
Frozen Foods              856
Dairy                     682
Canned                    649
Baking Goods              648
Health and Hygiene        520
Soft Drinks               445
Meat                      425
Breads                    251
Hard Drinks               214
Others                    169
Starchy Foods             148
Breakfast                 110
Seafood                    64
Name: Item_Type, dtype: int64

In [85]:
# Check for similar values in Item Fat Content
foodsales_df['Item_Fat_Content'].value_counts()

Low Fat    5089
Regular    2889
LF          316
reg         117
low fat     112
Name: Item_Fat_Content, dtype: int64

In [115]:
# Standardize the Low fat Data
#Adding LF to Low Fat
foodsales_df['Item_Fat_Content'] = foodsales_df['Item_Fat_Content'].replace('LF', 'Low Fat')

#Adding low fat to Low Fat
foodsales_df['Item_Fat_Content'] = foodsales_df['Item_Fat_Content'].replace('low fat', 'Low Fat')

In [116]:
# Standardize the Regular Data
#Adding reg to Regular

foodsales_df['Item_Fat_Content'] = foodsales_df['Item_Fat_Content'].replace('reg', 'Regular')


In [117]:
#Check to make sure
foodsales_df['Item_Fat_Content'].value_counts()

Low Fat    5517
Regular    3006
Name: Item_Fat_Content, dtype: int64

In [118]:
# Check for similar values in Outlet Location Type
foodsales_df['Outlet_Location_Type'].value_counts()

Tier 3    3350
Tier 2    2785
Tier 1    2388
Name: Outlet_Location_Type, dtype: int64

In [119]:
# Check for similar values in Outlet Type
foodsales_df['Outlet_Type'].value_counts()



Supermarket Type1    5577
Grocery Store        1083
Supermarket Type3     935
Supermarket Type2     928
Name: Outlet_Type, dtype: int64

Performing an analysis to determine all minimum, maximum, and mean for all values with numerical values

In [121]:
#Determine all numerical values
foodsales_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 10 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Fat_Content           8523 non-null   object 
 2   Item_Visibility            8523 non-null   float64
 3   Item_Type                  8523 non-null   object 
 4   Item_MRP                   8523 non-null   float64
 5   Outlet_Identifier          8523 non-null   object 
 6   Outlet_Establishment_Year  8523 non-null   int64  
 7   Outlet_Location_Type       8523 non-null   object 
 8   Outlet_Type                8523 non-null   object 
 9   Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(3), int64(1), object(6)
memory usage: 666.0+ KB


In [127]:
#Find the min , max and mean of Item Item_Visibility
print("For Item Visible")
print( "Minimum is ", round(foodsales_df['Item_Visibility'].min(), 3))
print( "Maximum is ", round(foodsales_df['Item_Visibility'].max(), 3))
print( "Mean is ", round(foodsales_df['Item_Visibility'].mean(), 3))

For Item Visible
Minimum is  0.0
Maximum is  0.328
Mean is  0.066


In [128]:
#Find the min , max and mean of Item Item_MRP
print("For Item MRP")
print( "Minimum is ", round(foodsales_df['Item_MRP'].min(), 2))
print( "Maximum is ", round(foodsales_df['Item_MRP'].max(), 2))
print( "Mean is ", round(foodsales_df['Item_MRP'].mean(), 2))

For Item MRP
Minimum is  31.29
Maximum is  266.89
Mean is  140.99


In [129]:
#Find the min , max and mean of Outlet Establishment Year
print("For Item Outlet Establishment Year")
print( "Minimum is ", round(foodsales_df['Outlet_Establishment_Year'].min()))
print( "Maximum is ", round(foodsales_df['Outlet_Establishment_Year'].max()))
print( "Mean is ", round(foodsales_df['Outlet_Establishment_Year'].mean()))

For Item Outlet Establishment Year
Minimum is  1985
Maximum is  2009
Mean is  1998


In [130]:
#Find the min , max and mean of Item Item_MRP
print("For Item Outlet Sales")
print( "Minimum is ", round(foodsales_df['Item_Outlet_Sales'].min(), 2))
print( "Maximum is ", round(foodsales_df['Item_Outlet_Sales'].max(), 2))
print( "Mean is ", round(foodsales_df['Item_Outlet_Sales'].mean(), 2))

For Item Outlet Sales
Minimum is  33.29
Maximum is  13086.96
Mean is  2181.29
